tools/autobuild/dynamo_s3.py

```diff
@@ -1,42 +1,30 @@
-import boto
-from boto.s3.key import Key
+import boto3
 import os
 import datetime
 from decimal import *
 
 def date_string():
-	return datetime.datetime.now().strftime("%Y%m%dT%H%M")
+    return datetime.datetime.now().strftime("%Y%m%dT%H%M")
 
 def mb(num_bytes):
+    return float(num_bytes) / 1000000
 
-	return float(num_bytes) / 1000000
-
-def report_progress( bytes_so_far, total_bytes ):
-	
-	print("Upload progress: %.2f mb / %.2f mb ( %.2f percent )" % (mb(bytes_so_far), mb(total_bytes), 100 * float(bytes_so_far)/total_bytes))
+def report_progress(bytes_so_far, total_bytes):
+    print("Upload progress: %.2f mb / %.2f mb ( %.2f percent )" % (mb(bytes_so_far), mb(total_bytes), 100 * float(bytes_so_far) / total_bytes))
 
 def upload_installer(fn, prefix, include_date, is_dev_build, extension, date_string):
+    s3 = boto3.client('s3')
 
-	s3 = boto.connect_s3()
-
-	if extension == '.exe' or extension == '.zip':
-		if is_dev_build:
-			b = s3.get_bucket('dyn-builds-dev')
-		else:
-			b = s3.get_bucket('dyn-builds-data')
-	elif extension == '.sig':
-		if is_dev_build:
-			b = s3.get_bucket('dyn-builds-dev-sig')
-		else:
-			b = s3.get_bucket('dyn-builds-data-sig')
-
-	k = Key(b)
-	
-	key = 'daily/' + prefix + '_' + date_string + extension if include_date else 'daily/' + prefix + extension
-	
-	k.key = os.path.basename( key )
-
-	k.set_contents_from_filename(fn, cb = report_progress, num_cb = 40)
-
+    if extension == '.exe' or extension == '.zip':
+        bucket_name = 'dyn-builds-dev' if is_dev_build else 'dyn-builds-data'
+    elif extension == '.sig':
+        bucket_name = 'dyn-builds-dev-sig' if is_dev_build else 'dyn-builds-data-sig'
 
+    key = 'daily/' + prefix + '_' + date_string + extension if include_date else 'daily/' + prefix + extension
 
+    s3.upload_file(
+        Filename=fn,
+        Bucket=bucket_name,
+        Key=os.path.basename(key),
```

Comment:
Hey @QilongTang, I had a question here, a few lines above we are prefixing the file name with other things, but here we are passing just the file name. Should we remove the  line above?

Reply:
The key seems to be the date string `20250327T0518` from e.g. `DynamoCoreRuntime_3.5.0.8161_20250327T0518.zip`. Maybe not change this for now

Reply:
Per our conversation at standup, feel free to make the change!